src/entry_point.py

```diff
@@ -201,6 +256,311 @@ def _constructor_menuinst(prefix, pkg_names=None, root_prefix=None, remove=False
+
+    from conda.base.constants import COMPATIBLE_SHELLS
+    from conda.common.compat import on_win
+    from conda.common.path import win_path_to_unix
+    from conda.core.initialize import (
+        CONDA_INITIALIZE_PS_RE_BLOCK,
+        CONDA_INITIALIZE_RE_BLOCK,
+        _read_windows_registry,
+        make_initialize_plan,
+    )
+
+    BIN_DIRECTORY = "Scripts" if on_win else "bin"
+    reverse_plan = []
+    for shell in COMPATIBLE_SHELLS:
+        # Make plan for each shell individually because
+        # not every plan includes the shell name
+        plan = make_initialize_plan(
+            str(uninstall_prefix),
+            [shell],
+            for_user,
+            for_system,
+            anaconda_prompt,
+            reverse=True,
+        )
+        for initializer in plan:
+            target_path = initializer["kwargs"]["target_path"]
+            if target_path.startswith("HKEY"):
+                # target_path for cmd.exe is a registry path
+                reg_entry, _ = _read_windows_registry(target_path)
+                if not isinstance(reg_entry, str):
+                    continue
+                autorun_parts = reg_entry.split("&")
+                for prefix in prefixes:
+                    hook = str(prefix / "condabin" / "conda_hook.bat")
+                    if any(hook in part for part in autorun_parts):
+                        reverse_plan.append(initializer)
+                        break
+            else:
+                target_path = Path(target_path)
+                # Only reverse for paths that are outside the uninstall prefix
+                # since paths inside the uninstall prefix will be deleted anyway
+                if not target_path.exists() or _is_subdir(
+                    target_path, uninstall_prefix
+                ):
+                    continue
+                rc_content = target_path.read_text()
+                if shell == "powershell":
+                    pattern = CONDA_INITIALIZE_PS_RE_BLOCK
+                else:
+                    pattern = CONDA_INITIALIZE_RE_BLOCK
+                flags = re.MULTILINE
+                matches = re.findall(pattern, rc_content, flags=flags)
+                if not matches:
+                    continue
+                for prefix in prefixes:
+                    # Ignore .exe suffix to make the logic simpler
+                    if shell in ("csh", "tcsh") and sys.platform != "win32":
+                        sentinel_str = str(prefix / "etc" / "profile.d" / "conda.csh")
+                    else:
+                        sentinel_str = str(prefix / BIN_DIRECTORY / "conda")
+                    if sys.platform == "win32" and shell != "powershell":
+                        sentinel_str = win_path_to_unix(sentinel_str)
+                        # Remove /cygdrive to make the path shell-independent
+                        if sentinel_str.startswith("/cygdrive"):
+                            sentinel_str = sentinel_str[9:]
```

Comment:
You can use https://docs.python.org/3/library/stdtypes.html#str.removeprefix now :D